src/routes/credentials.rs

```diff
@@ -282,23 +295,78 @@ async fn post_credential_handler(
         .get(&oidc_context.client_id)
         .ok_or(Error::OauthInvalidClientId)?;
     if let Some(checks_directory) = &client_config.checks_directory {
-        let checker = app_state
+        let mut app_state_write = app_state.write().await;
+        let checker = app_state_write
             .rhai_checkers
             .get_or_create(&oidc_context.client_id, checks_directory)?;
         checker.check(&id_token)?;
+        drop(app_state_write);
```

Comment:
This is `drop`ped automatically once the variable goes of out scope at the end of the block, so we don't need that.